assets/scripts/info_bubble/InfoBubble.jsx

```diff
@@ -0,0 +1,208 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import { connect } from 'react-redux'
+import RemoveButton from './RemoveButton'
+import Variants from './Variants'
+import WidthControl from './WidthControl'
+import BuildingHeightControl from './BuildingHeightControl'
+import Warnings from './Warnings'
+import Description from './Description.jsx'
+import { infoBubble } from './info_bubble'
+import { getDescriptionData } from './description'
+import { resumeFadeoutControls } from '../segments/resizing'
+import { getStreet } from '../streets/data_model'
+// import { trackEvent } from '../app/event_tracking'
+import { BUILDING_VARIANTS, BUILDING_VARIANT_NAMES } from '../segments/buildings'
+import { SEGMENT_INFO } from '../segments/info'
+import { loseAnyFocus } from '../util/focus'
+import { setInfoBubbleMouseInside } from '../store/actions/infoBubble'
+import { t } from '../app/locale'
+
+const INFO_BUBBLE_TYPE_SEGMENT = 1
+const INFO_BUBBLE_TYPE_LEFT_BUILDING = 2
+const INFO_BUBBLE_TYPE_RIGHT_BUILDING = 3
+
+class InfoBubble extends React.Component {
+  static propTypes = {
+    visible: PropTypes.bool.isRequired,
+    dataNo: PropTypes.number,
+    setInfoBubbleMouseInside: PropTypes.func
+  }
+
+  static defaultProps = {
+    visible: false
+  }
+
+  constructor (props) {
+    super(props)
+
+    this.state = {
+      type: null,
+      street: null,
+      segment: null,
+      description: null
+    }
+  }
+
+  componentDidMount () {
+    // This listener hides the info bubble when the mouse leaves the
+    // document area. Do not normalize it to a pointerleave event
+    // because it doesn't make sense for other pointer types
+    // document.addEventListener('mouseleave', infoBubble.hide)
```

Comment:
Did you mean to leave this line commented?

Reply:
I did not. Good catch!